### PR TITLE
Feat(eos_designs): Add sample rate to sflow settings in eos_designs.

### LINKED ISSUE #3802: Add sample rate to sflow settings in eos_designs
### Enhancement summary

As discussed with @gmuloc, we are trying to leverage on eos_designs for most of our global vars.

However, sflow sample rate seems is not something that can be configured today via eos_designs:
https://avd.sh/en/stable/roles/eos_designs/docs/input-variables.html#sflow-settings

This is only configurable via eos_cli_config_gen:
https://avd.sh/en/stable/roles/eos_cli_config_gen/docs/input-variables.html#sflow

### Which component of AVD is impacted

eos_designs

### Use case example

Add support for sample rates, which are commonly modified by customers.

### Describe the solution you would like

Something along the following:
```
sflow_settings:
  sample: <int>
```

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

 Add sample rate to sflow settings in eos_designs.

## Related Issue(s)

Fixes #3802 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Adding `sample` key to `sflow_settings` data-model in eos_designs
```

sflow_settings:
     sample:
          rate: <int; 1-4294967295>
```

## How to test

Test written in molecule scenario `eos_designs_unit_tests`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml [hunk 1]
```diff
@@ -15,6 +15,13 @@ keys:
       The sFlow process will only be configured if any interface is enabled for sFlow.
       For default enabling of sFlow for various interface types across the fabric see `fabric_sflow`.
     keys:
+      sample:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml:18]:
```suggestion
      sample_rate:
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml [hunk 2]
```diff
@@ -15,6 +15,13 @@ keys:
...
+        type: int
+        convert_types:
+          - str
+        min: 1
+        max: 4294967295
+        description: Sample rate.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml]: Please add more details about the rate. Is it 1:<sample rate> packets?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/flows/avdstructuredconfig.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def sflow(self) -> dict | None:
         # and at least one destination.
         sflow = {"run": True}
 
+        # Add the packet sampling rate.
+        sflow["sample"] = get(self._hostvars, "sflow_settings.sample", default=None)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/flows/avdstructuredconfig.py]:
```suggestion
        sflow = {
            "run": True,
            "sample": get(self._hostvars, "sflow_settings.sample_rate"),
        }        
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml (hunk 1 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml:18]: There are few more keys inside sample, so I have restructured the data-model now.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml (hunk 2 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/sflow_settings.schema.yml]: Added

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/flows/avdstructuredconfig.py (hunk 3 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/flows/avdstructuredconfig.py]: Applied

**Review by ClausHolbechArista — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Pull request merged**